File: PhongTroAPI/PhongTroAPI/Entities/Phong.cs
using System;
using System.Collections.Generic;

namespace PhongTroAPI.Entities;

public partial class Phong
{
    public int MaPhong { get; set; }

    public int MaCoSo { get; set; }

    public string SoPhong { get; set; } = null!;

    public int? Tang { get; set; }

    public double? DienTich { get; set; }

    public decimal GiaThue { get; set; }

    public int? SoNguoiToiDa { get; set; }

    public string? TrangThai { get; set; }

    public string? MoTa { get; set; }

    public DateTime? NgayTao { get; set; }

    public virtual ICollection<HinhAnhPhong> HinhAnhPhongs { get; set; } = new List<HinhAnhPhong>();

    public virtual ICollection<HoaDon> HoaDons { get; set; } = new List<HoaDon>();

    public virtual ICollection<HopDongThue> HopDongThues { get; set; } = new List<HopDongThue>();

    public virtual CoSo MaCoSoNavigation { get; set; } = null!;

    public virtual ICollection<TienIch> MaTienIches { get; set; } = new List<TienIch>();
}

File: trosmart/lib/models/phong.dart
class Phong {
  final int maPhong;
  final int maCoSo;
  final String soPhong;
  final int? tang;
  final double? dienTich;
  final double giaThue;
  final int? soNguoiToiDa;
  final String? trangThai;
  final String? moTa;
  final DateTime? ngayTao;

  Phong({
    required this.maPhong,
    required this.maCoSo,
    required this.soPhong,
    this.tang,
    this.dienTich,
    required this.giaThue,
    this.soNguoiToiDa,
    this.trangThai,
    this.moTa,
    this.ngayTao,
  });

  factory Phong.fromJson(Map<String, dynamic> json) {
    return Phong(
      maPhong: json['maPhong'],
      maCoSo: json['maCoSo'],
      soPhong: json['soPhong'] ?? '',
      tang: json['tang'],
      dienTich: json['dienTich']?.toDouble(),
      giaThue: json['giaThue']?.toDouble() ?? 0.0,
      soNguoiToiDa: json['soNguoiToiDa'],
      trangThai: json['trangThai'],
      moTa: json['moTa'],
      ngayTao: json['ngayTao'] != null ? DateTime.tryParse(json['ngayTao']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'maPhong': maPhong,
      'maCoSo': maCoSo,
      'soPhong': soPhong,
      'tang': tang,
      'dienTich': dienTich,
      'giaThue': giaThue,
      'soNguoiToiDa': soNguoiToiDa,
      'trangThai': trangThai,
      'moTa': moTa,
      'ngayTao': ngayTao?.toIso8601String(),
    };
  }
}
